compiler/optimizer/LoopVersioner.cpp

```diff
@@ -187,7 +187,9 @@ bool TR_LoopVersioner::loopIsWorthVersioning(TR_RegionStructure *naturalLoop)
    if (entryBlock->isCold())
       return false;
 
-   if (comp()->getMethodHotness() <= warm)
+   // if aggressive loop versioner flag is set then run at any optlevel, otherwise only at warm or less
+   bool aggressive = TR::Options::getCmdLineOptions()->getOption(TR_EnableAggressiveLoopVersioning);
+   if (aggressive || comp()->getMethodHotness() <= warm)
```

Comment:
This probably deserves a trace message to show the decision made - it should be guarded for tracing of the opt in question.